File: 0083.RemoveDuplicatesfromSortedList.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        p = head
        while p:
            while p.next and p.next.val == p.val:
                p.next = p.next.next
            p = p.next
        return head
            

File: 0083.RemoveDuplicatesfromSortedList.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func deleteDuplicates(head *ListNode) *ListNode {
    p := head
    for p != nil {
        tmp := p.Next
        for tmp != nil && tmp.Val == p.Val {
            tmp = tmp.Next
        }
        p.Next = tmp
        p = tmp
    }
    return head
}